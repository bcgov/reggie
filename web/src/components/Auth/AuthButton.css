.auth-button {
  position: relative;
  margin-left: auto;
  padding: 8px 16px;
  border: none;
  background-color: #fcba19;
  color: #003366;
  text-transform: uppercase;
  letter-spacing: 0.2em;
  border-radius: 2px;
  outline-width: 0;
  text-decoration: none;
  overflow: hidden;
  cursor: pointer;
  -webkit-transition-duration: 0.4s; /* Safari */
  transition-duration: 0.4s;
}

@media screen and (max-width: 500px) {
  .auth-button {
    letter-spacing: .1em;
    padding: 4px 8px;
    font-size: 0.8rem;
  }
}
