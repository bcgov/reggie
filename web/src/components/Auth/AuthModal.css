.auth-modal.modal {
  display: flex !important;
  align-items: center;
  background-color: white;
}

.modal-header {
  background-color: #003366;
  border-bottom: 2px solid #fcba19;
  color: #fff;
  height: 100px;
  padding: 0.5rem 1rem !important;
}

.modal-title {
  width: 400px;
}

.auth-modal .modal-footer {
  display: block;
}

.auth-modal .auth-buttons {
  display: flex;
  justify-content: space-evenly;
}

.auth-modal .auth-button {
  margin: 0;
}

.modal-body {
  height: 100px;
}

@media screen and (max-width: 500px) {
  .auth-modal .modal-dialog {
    width: 100%;
    margin: 0;
  }
}
