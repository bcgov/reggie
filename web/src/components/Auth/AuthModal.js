//
// Reggie Web
//
// Copyright Â© 2018 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Jason Leach on 2018-10-03.
//

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import Banner from '../UI/Banner';
import { SSO_IDP, APP_INFO } from '../../constants';
import './AuthModal.css';

export const AuthModal = ({ isAuthenticated }) => {
  return (
    <Modal modalClassName="auth-modal" isOpen={!isAuthenticated} fade={false}>
      <ModalHeader>
        <Banner titleText={APP_INFO.NAME} />
      </ModalHeader>
      <ModalBody>Please login before starting Rocket.Chat Invitation with Reggie</ModalBody>
      <ModalFooter>
        <p>Login with:</p>
        <div className="auth-buttons">
          <Link className="auth-button" to={{ pathname: '/login', state: { idp: SSO_IDP.GITHUB } }}>
            GitHub
          </Link>
          <Link className="auth-button" to={{ pathname: '/login', state: { idp: SSO_IDP.IDIR } }}>
            IDIR
          </Link>
        </div>
      </ModalFooter>
    </Modal>
  );
};

AuthModal.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
};

export default AuthModal;
