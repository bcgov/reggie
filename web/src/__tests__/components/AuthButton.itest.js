import React from 'react';
import { shallow, mount } from 'enzyme';
import { AuthButton } from '../../components/Auth/AuthButton';

describe.skip('AuthButton Component', () => {
  it('matches snapshot when Logged in', () => {
    const authState = true;
    const wrapper = shallow(<AuthButton isAuthenticated={authState} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when Logged out', () => {
    const authState = false;
    const wrapper = shallow(<AuthButton isAuthenticated={authState} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('redirect to log in url after clicking', () => {
    window.location.assign = jest.fn();
    const authState = false;
    const wrapper = mount(<AuthButton isAuthenticated={authState} />);
    expect(wrapper.prop('isAuthenticated')).toEqual(authState);
    wrapper
      .find('.auth-button')
      .first()
      .simulate('click');
    expect(window.location.assign).toHaveBeenCalled();
  });

  it('redirect to log out url after clicking', () => {
    window.location.assign = jest.fn();
    const authState = true;
    const wrapper = mount(<AuthButton isAuthenticated={authState} />);
    expect(wrapper.prop('isAuthenticated')).toEqual(authState);
    wrapper
      .find('.auth-button')
      .first()
      .simulate('click');
    expect(window.location.assign).toHaveBeenCalled();
  });
});
