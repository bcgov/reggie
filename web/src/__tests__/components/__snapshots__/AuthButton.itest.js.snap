// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton Component matches snapshot when Logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthButton
    isAuthenticated={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="auth-button"
        replace={false}
        to="/logout"
      >
        Logout
      </Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Logout",
        "className": "auth-button",
        "replace": false,
        "to": "/logout",
      },
      "ref": null,
      "rendered": "Logout",
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="auth-button"
          replace={false}
          to="/logout"
        >
          Logout
        </Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Logout",
          "className": "auth-button",
          "replace": false,
          "to": "/logout",
        },
        "ref": null,
        "rendered": "Logout",
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`AuthButton Component matches snapshot when Logged out 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthButton
    isAuthenticated={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="auth-button"
        replace={false}
        to="/login"
      >
        Login
      </Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Login",
        "className": "auth-button",
        "replace": false,
        "to": "/login",
      },
      "ref": null,
      "rendered": "Login",
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="auth-button"
          replace={false}
          to="/login"
        >
          Login
        </Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Login",
          "className": "auth-button",
          "replace": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
